### DISCUSSION TITLE
菜单项过多时，二级菜单覆盖在一级菜单上

### DISCUSSION THREAD

**Discussion by dongleizi** [element-plus/element-plus#11427] [Q&A]
菜单项过多时，二级菜单覆盖在一级菜单上，又遇到这种问题的吗？该如何解决？

[演示地址](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtcmFkaW8tZ3JvdXAgdi1tb2RlbD1cImlzQ29sbGFwc2VcIiBzdHlsZT1cIm1hcmdpbi1ib3R0b206IDIwcHhcIj5cbiAgICA8ZWwtcmFkaW8tYnV0dG9uIDpsYWJlbD1cImZhbHNlXCI+ZXhwYW5kPC9lbC1yYWRpby1idXR0b24+XG4gICAgPGVsLXJhZGlvLWJ1dHRvbiA6bGFiZWw9XCJ0cnVlXCI+Y29sbGFwc2U8L2VsLXJhZGlvLWJ1dHRvbj5cbiAgPC9lbC1yYWRpby1ncm91cD5cbiAgPGVsLW1lbnVcbiAgICBkZWZhdWx0LWFjdGl2ZT1cIjJcIlxuICAgIGNsYXNzPVwiZWwtbWVudS12ZXJ0aWNhbC1kZW1vXCJcbiAgICA6Y29sbGFwc2U9XCJpc0NvbGxhcHNlXCJcbiAgICBAb3Blbj1cImhhbmRsZU9wZW5cIlxuICAgIEBjbG9zZT1cImhhbmRsZUNsb3NlXCJcbiAgPlxuICAgIDxlbC1zdWItbWVudSBpbmRleD1cIjFcIj5cbiAgICAgIDx0ZW1wbGF0ZSAjdGl0bGU+XG4gICAgICAgIDxlbC1pY29uPjxsb2NhdGlvbiAvPjwvZWwtaWNvbj5cbiAgICAgICAgPHNwYW4+TmF2aWdhdG9yIE9uZTwvc3Bhbj5cbiAgICAgIDwvdGVtcGxhdGU+XG4gICAgICA8ZWwtbWVudS1pdGVtLWdyb3VwPlxuICAgICAgICA8dGVtcGxhdGUgI3RpdGxlPjxzcGFuPkdyb3VwIE9uZTwvc3Bhbj48L3RlbXBsYXRlPlxuICAgICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiMS0xXCI+aXRlbSBvbmU8L2VsLW1lbnUtaXRlbT5cbiAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjEtMlwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIxLTEyMlwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIxLTIzXCI+aXRlbSB0d288L2VsLW1lbnUtaXRlbT5cbiAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjEtMjJcIj5pdGVtIHR3bzwvZWwtbWVudS1pdGVtPlxuICAgICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiMS0xMlwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIxLTEyMVwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIxLTEyNFwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIxLTEyMlwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIxLTEyMVwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIxLTIxXCI+aXRlbSB0d288L2VsLW1lbnUtaXRlbT5cbiAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjEtMjQxXCI+aXRlbSB0d288L2VsLW1lbnUtaXRlbT5cbiAgICAgIDwvZWwtbWVudS1pdGVtLWdyb3VwPlxuICAgICAgPGVsLW1lbnUtaXRlbS1ncm91cCB0aXRsZT1cIkdyb3VwIFR3b1wiPlxuICAgICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiMS0zXCI+aXRlbSB0aHJlZTwvZWwtbWVudS1pdGVtPlxuICAgICAgPC9lbC1tZW51LWl0ZW0tZ3JvdXA+XG4gICAgICA8ZWwtc3ViLW1lbnUgaW5kZXg9XCIxLTRcIj5cbiAgICAgICAgPHRlbXBsYXRlICN0aXRsZT48c3Bhbj5pdGVtIGZvdXI8L3NwYW4+PC90ZW1wbGF0ZT5cbiAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjEtNC0xXCI+aXRlbSBvbmU8L2VsLW1lbnUtaXRlbT5cbiAgICAgIDwvZWwtc3ViLW1lbnU+XG4gICAgPC9lbC1zdWItbWVudT5cbiAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiMlwiPlxuICAgICAgPGVsLWljb24+PGljb24tbWVudSAvPjwvZWwtaWNvbj5cbiAgICAgIDx0ZW1wbGF0ZSAjdGl0bGU+TmF2aWdhdG9yIFR3bzwvdGVtcGxhdGU+XG4gICAgPC9lbC1tZW51LWl0ZW0+XG4gICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjNcIiBkaXNhYmxlZD5cbiAgICAgIDxlbC1pY29uPjxkb2N1bWVudCAvPjwvZWwtaWNvbj5cbiAgICAgIDx0ZW1wbGF0ZSAjdGl0bGU+TmF2aWdhdG9yIFRocmVlPC90ZW1wbGF0ZT5cbiAgICA8L2VsLW1lbnUtaXRlbT5cbiAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiNFwiPlxuICAgICAgPGVsLWljb24+PHNldHRpbmcgLz48L2VsLWljb24+XG4gICAgICA8dGVtcGxhdGUgI3RpdGxlPk5hdmlnYXRvciBGb3VyPC90ZW1wbGF0ZT5cbiAgICA8L2VsLW1lbnUtaXRlbT5cbiAgPC9lbC1tZW51PlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7XG4gIERvY3VtZW50LFxuICBNZW51IGFzIEljb25NZW51LFxuICBMb2NhdGlvbixcbiAgU2V0dGluZyxcbn0gZnJvbSAnQGVsZW1lbnQtcGx1cy9pY29ucy12dWUnXG5cbmNvbnN0IGlzQ29sbGFwc2UgPSByZWYodHJ1ZSlcbmNvbnN0IGhhbmRsZU9wZW4gPSAoa2V5OiBzdHJpbmcsIGtleVBhdGg6IHN0cmluZ1tdKSA9PiB7XG4gIGNvbnNvbGUubG9nKGtleSwga2V5UGF0aClcbn1cbmNvbnN0IGhhbmRsZUNsb3NlID0gKGtleTogc3RyaW5nLCBrZXlQYXRoOiBzdHJpbmdbXSkgPT4ge1xuICBjb25zb2xlLmxvZyhrZXksIGtleVBhdGgpXG59XG48L3NjcmlwdD5cblxuPHN0eWxlPlxuLmVsLW1lbnUtdmVydGljYWwtZGVtbzpub3QoLmVsLW1lbnUtLWNvbGxhcHNlKSB7XG4gIHdpZHRoOiAyMDBweDtcbiAgbWluLWhlaWdodDogNDAwcHg7XG59XG48L3N0eWxlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319)

![image](https://user-images.githubusercontent.com/13233534/216864512-2f9b8a6b-0f19-40d3-85db-177552bf6766.png)

**Comment by chenxch** [comment]
[playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtcmFkaW8tZ3JvdXAgdi1tb2RlbD1cImlzQ29sbGFwc2VcIiBzdHlsZT1cIm1hcmdpbi1ib3R0b206IDIwcHhcIj5cbiAgICA8ZWwtcmFkaW8tYnV0dG9uIDpsYWJlbD1cImZhbHNlXCI+ZXhwYW5kPC9lbC1yYWRpby1idXR0b24+XG4gICAgPGVsLXJhZGlvLWJ1dHRvbiA6bGFiZWw9XCJ0cnVlXCI+Y29sbGFwc2U8L2VsLXJhZGlvLWJ1dHRvbj5cbiAgPC9lbC1yYWRpby1ncm91cD5cbiAgPGVsLW1lbnVcbiAgICBkZWZhdWx0LWFjdGl2ZT1cIjJcIlxuICAgIGNsYXNzPVwiZWwtbWVudS12ZXJ0aWNhbC1kZW1vXCJcbiAgICA6Y29sbGFwc2U9XCJpc0NvbGxhcHNlXCJcbiAgICBAb3Blbj1cImhhbmRsZU9wZW5cIlxuICAgIEBjbG9zZT1cImhhbmRsZUNsb3NlXCJcbiAgPlxuICAgIDxlbC1zdWItbWVudSBpbmRleD1cIjFcIj5cbiAgICAgIDx0ZW1wbGF0ZSAjdGl0bGU+XG4gICAgICAgIDxlbC1pY29uPjxsb2NhdGlvbiAvPjwvZWwtaWNvbj5cbiAgICAgICAgPHNwYW4+TmF2aWdhdG9yIE9uZTwvc3Bhbj5cbiAgICAgIDwvdGVtcGxhdGU+XG4gICAgICA8ZGl2IHN0eWxlPVwibWF4LWhlaWdodDogNTAwcHg7IG92ZXJmbG93OiBhdXRvO1wiPlxuICAgICAgICAgICAgICA8ZWwtbWVudS1pdGVtLWdyb3VwPlxuICAgICAgICA8dGVtcGxhdGUgI3RpdGxlPjxzcGFuPkdyb3VwIE9uZTwvc3Bhbj48L3RlbXBsYXRlPlxuICAgICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiMS0xXCI+aXRlbSBvbmU8L2VsLW1lbnUtaXRlbT5cbiAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjEtMlwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIxLTEyMlwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIxLTIzXCI+aXRlbSB0d288L2VsLW1lbnUtaXRlbT5cbiAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjEtMjJcIj5pdGVtIHR3bzwvZWwtbWVudS1pdGVtPlxuICAgICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiMS0xMlwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIxLTEyMVwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIxLTEyNFwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIxLTEyMlwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIxLTEyMVwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIxLTIxXCI+aXRlbSB0d288L2VsLW1lbnUtaXRlbT5cbiAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjEtMjQxXCI+aXRlbSB0d288L2VsLW1lbnUtaXRlbT5cbiAgICAgIDwvZWwtbWVudS1pdGVtLWdyb3VwPlxuICAgICAgPGVsLW1lbnUtaXRlbS1ncm91cCB0aXRsZT1cIkdyb3VwIFR3b1wiPlxuICAgICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiMS0zXCI+aXRlbSB0aHJlZTwvZWwtbWVudS1pdGVtPlxuICAgICAgPC9lbC1tZW51LWl0ZW0tZ3JvdXA+XG4gICAgICA8ZWwtc3ViLW1lbnUgaW5kZXg9XCIxLTRcIj5cbiAgICAgICAgPHRlbXBsYXRlICN0aXRsZT48c3Bhbj5pdGVtIGZvdXI8L3NwYW4+PC90ZW1wbGF0ZT5cbiAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjEtNC0xXCI+aXRlbSBvbmU8L2VsLW1lbnUtaXRlbT5cbiAgICAgIDwvZWwtc3ViLW1lbnU+XG4gICAgICA8L2Rpdj5cblxuICAgIDwvZWwtc3ViLW1lbnU+XG4gICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjJcIj5cbiAgICAgIDxlbC1pY29uPjxpY29uLW1lbnUgLz48L2VsLWljb24+XG4gICAgICA8dGVtcGxhdGUgI3RpdGxlPk5hdmlnYXRvciBUd288L3RlbXBsYXRlPlxuICAgIDwvZWwtbWVudS1pdGVtPlxuICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIzXCIgZGlzYWJsZWQ+XG4gICAgICA8ZWwtaWNvbj48ZG9jdW1lbnQgLz48L2VsLWljb24+XG4gICAgICA8dGVtcGxhdGUgI3RpdGxlPk5hdmlnYXRvciBUaHJlZTwvdGVtcGxhdGU+XG4gICAgPC9lbC1tZW51LWl0ZW0+XG4gICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjRcIj5cbiAgICAgIDxlbC1pY29uPjxzZXR0aW5nIC8+PC9lbC1pY29uPlxuICAgICAgPHRlbXBsYXRlICN0aXRsZT5OYXZpZ2F0b3IgRm91cjwvdGVtcGxhdGU+XG4gICAgPC9lbC1tZW51LWl0ZW0+XG4gIDwvZWwtbWVudT5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5pbXBvcnQge1xuICBEb2N1bWVudCxcbiAgTWVudSBhcyBJY29uTWVudSxcbiAgTG9jYXRpb24sXG4gIFNldHRpbmcsXG59IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBpc0NvbGxhcHNlID0gcmVmKHRydWUpXG5jb25zdCBoYW5kbGVPcGVuID0gKGtleTogc3RyaW5nLCBrZXlQYXRoOiBzdHJpbmdbXSkgPT4ge1xuICBjb25zb2xlLmxvZyhrZXksIGtleVBhdGgpXG59XG5jb25zdCBoYW5kbGVDbG9zZSA9IChrZXk6IHN0cmluZywga2V5UGF0aDogc3RyaW5nW10pID0+IHtcbiAgY29uc29sZS5sb2coa2V5LCBrZXlQYXRoKVxufVxuPC9zY3JpcHQ+XG5cbjxzdHlsZT5cbi5lbC1tZW51LXZlcnRpY2FsLWRlbW86bm90KC5lbC1tZW51LS1jb2xsYXBzZSkge1xuICB3aWR0aDogMjAwcHg7XG4gIG1pbi1oZWlnaHQ6IDQwMHB4O1xufVxuPC9zdHlsZT5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)
可以尝试套一层div，设置max-height和overflow
